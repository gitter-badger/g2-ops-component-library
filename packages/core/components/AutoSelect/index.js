// @flow
export {AutoSelect} from './AutoSelect'
