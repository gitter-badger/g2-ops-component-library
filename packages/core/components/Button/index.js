export { IconButton, Button } from './Button'
