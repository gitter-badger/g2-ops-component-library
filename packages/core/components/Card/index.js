export { Card, CardActions, CardHeader, CardMedia, CardTitle, CardText } from './Card'
