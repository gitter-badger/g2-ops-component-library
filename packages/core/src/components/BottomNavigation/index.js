// export { BottomNavigation, BottomNavigationItem } from './BottomNavigation'
export { BottomNavigation, BottomNavigationItem } from './BottomNavigation'
