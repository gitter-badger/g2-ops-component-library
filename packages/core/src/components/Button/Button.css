.copartButton {
	height: 36px;
	min-width: 110px !important;
}

.copartButton button:disabled {
	cursor: not-allowed !important;
}

.copartButton button span {
	font-weight: 500 !important;
	text-transform: capitalize !important;
}

.primaryButton button {
	background-color: #1d5ab9 !important;
}

.primaryButton button:hover {
	background-color: #133a77 !important;
}

.primaryButton button span {
	color: white !important;
}

.secondaryButton button {
	background-color: white !important;
}

.secondaryButton button:hover {
	background-color: #d9d9d9 !important;
}

.inactiveButton button {
	background-color: #dfdfdf !important;
}

.inactiveButton button:hover {
	background-color: #b9b9b9 !important;
}

.deleteButton button {
	background-color: #ee2727 !important;
}

.deleteButton button:hover {
	background-color: #ba0f0f !important;
}

.deleteButton button span {
	color: white !important;
}

.addButton button svg, .editButton button svg, .deleteButton button svg {
	margin: 0 !important;
	color: white !important;
	fill: white !important;
}

.addButton button span, .editButton button span, .deleteButton button span {
	padding-left: 3px !important;
}

.addButton button svg, .deleteButton button svg {
	height: 20px !important;
	width: 20px !important;
}

.editButton button svg {
	height: 16px !important;
	width: 16px !important;
}