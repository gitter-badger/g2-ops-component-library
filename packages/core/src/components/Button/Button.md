Button Component Examples:
```jsx
<div>
  <span style={{ margin: '10px' }}><Button type="primary" label="Copart Primary Button" /></span>
  <span style={{ margin: '10px' }}><Button type="secondary" label="Secondary Button" /></span>
  <span style={{ margin: '10px' }}><Button type="inactive" disabled label="Inactive" /></span>
  <span style={{ margin: '10px' }}><Button type="add" /></span>
  <span style={{ margin: '10px' }}><Button type="edit" /></span>
  <span style={{ margin: '10px' }}><Button type="delete" /></span>
</div>
```

Icon Button Examples:

Render any icon from material icons that can be found here: https://material.io/icons/
```js
const { IconButton } = require('./Button');
const iconsList = [
  'save', 'delete_forever', 'close',
  'assignment', 'autorenew', 'cached',
  'call', 'cancel', 'create', 'exit_to_app',
  'favorite', 'filter_list', 'forward',
  'help', 'lock_outline', 'offline_pin',
  'open_in_new', 'phone', 'print', 'receipt',
  'replay', 'search', 'settings_backup_restore',
  'settings', 'sort', 'swap_horiz', 'add_circle',
  'remove_circle'
];
<div>
  {iconsList.map(iconName => (
    <IconButton
      key={iconName}
      tooltip={iconName}
      onClick={() => console.log(`${iconName} clicked.`)}>
      <i className="material-icons">{iconName}</i>
    </IconButton>
  ))}
</div>
```