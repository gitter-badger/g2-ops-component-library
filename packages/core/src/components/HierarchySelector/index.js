// @flow

export { HierarchySelector } from './HierarchySelector'
