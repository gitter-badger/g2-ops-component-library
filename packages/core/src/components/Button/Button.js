// @flow
import React from 'react'
import RaisedButton from 'material-ui/RaisedButton'
import MuiIconButton from 'material-ui/IconButton'
import AddIcon from 'material-ui/svg-icons/content/add'
import EditIcon from 'material-ui/svg-icons/image/edit'
import DeleteIcon from 'material-ui/svg-icons/action/delete-forever'

import { wrapMuiContext } from 'utilities/wrapMuiContext'

// TODO: Fix button + icon styling.
import './Button.css'

type ButtonPropsT = {
  /** To indicate type of button {primary, secondary, inactive, add, edit, delete} */
  variant: 'primary' | 'secondary' | 'inactive' | 'add' | 'edit' | 'delete',
  /** OnClick event for button */
  onClick: (SyntheticMouseEvent<HTMLInputElement>) => void,
}

export const Button = wrapMuiContext(({ variant, ...buttonProps }: ButtonPropsT) => {
	let typeProps = {}

  switch (variant) {
    case 'primary':
    case 'secondary':
    case 'inactive':
      typeProps = {
        ...buttonProps,
        className: `copartButton ${variant}Button ${buttonProps.className}`,
      }
      break


    case 'add':
      typeProps = {
        ...buttonProps,
        label: 'Add',
        className: `copartButton primaryButton ${variant}Button ${buttonProps.className}`,
        icon: <AddIcon />,
      }
      break

    case 'edit':
      typeProps = {
        ...buttonProps,
        label: 'Edit',
        className: `copartButton primaryButton ${variant}Button ${buttonProps.className}`,
        icon: <EditIcon />,
      }
      break

    case 'delete':
      typeProps = {
        ...buttonProps,
        label: 'Delete',
        className: `copartButton ${variant}Button ${buttonProps.className}`,
        icon: <DeleteIcon />,
      }
      break
    
    default:
      typeProps = { ...buttonProps }

  }

  return <RaisedButton {...typeProps} data-core-component="Button" />
})

Button.defaultProps = {
	className: ''
}

export const IconButton = wrapMuiContext(MuiIconButton)


