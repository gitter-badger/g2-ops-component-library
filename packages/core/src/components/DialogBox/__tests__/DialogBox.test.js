import React from 'react'
import renderer from 'react-test-renderer'

import {Button} from 'components/Button'

import {DialogBox} from '../'

// TODO: We don't need to test anything here!
describe('<DialogBox />', () => {
  test('should render properly', () => {
    const tree = renderer.create(
      <DialogBox
        title={'Dialog with Content'}
        hideDialog={false}
        dialogType="normal"
        onDismiss={() => {}}
        footerRenderer={() => (
          <div style={{ display: 'flex', float: 'right' }}>
            <Button variant='primary' onClick={() => {}} label={'Save'} />
            <span style={{ width: '10px' }}>{}</span>
            <Button variant='secondary' onClick={() => {}} label={'Discard'} />
          </div>
        )}
      >
        <div>Main Dialog Content to be added here</div>
      </DialogBox>,
    )
    expect(tree).toMatchSnapshot()
  })
})
