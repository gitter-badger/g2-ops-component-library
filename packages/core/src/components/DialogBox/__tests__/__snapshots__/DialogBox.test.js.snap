// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DialogBox /> should render properly 1`] = `
<div
  className="ms-Fabric root-69"
>
  <span
    className="ms-Layer"
  />
</div>
`;
